packages/@react-spectrum/tree/src/TreeView.tsx

```diff
@@ -0,0 +1,361 @@
+    'drag-handle checkbox level-padding expand-button icon content actions actionmenu'
+  ],
+  color: {
+    isDisabled: {
+      default: 'gray-400',
+      forcedColors: 'GrayText'
+    }
+  }
+});
+
+// TODO: These styles lose against the spectrum class names, so I've did unsafe for the ones that get overridden
+const treeCheckbox = style({
+  gridArea: 'checkbox',
+  transitionDuration: '160ms',
+  marginStart: 3,
+  marginEnd: 0
+});
+
+const treeIcon = style({
+  gridArea: 'icon',
+  marginEnd: 2
+});
+
+const treeContent = style<Pick<TreeItemContentRenderProps, 'isDisabled'>>({
+  gridArea: 'content',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden'
+});
+
+const treeActions = style({
+  gridArea: 'actions',
+  flexGrow: 0,
+  flexShrink: 0,
+  /* TODO: I made this one up, confirm desired behavior. These paddings are to make sure the action group has enough padding for the focus ring */
+  marginStart: .5,
+  marginEnd: 1
+});
+
+const treeActionMenu = style({
+  gridArea: 'actionmenu',
+  width: 8
+});
+
+const treeRowOutline = style({
+  content: '',
+  display: 'block',
+  position: 'absolute',
+  insetStart: 0,
+  insetEnd: 0,
+  top: {
+    default: 0,
+    isFocusVisible: '[-2px]',
+    isSelected: {
+      default: '[-1px]',
+      isFocusVisible: '[-2px]'
+    }
+  },
+  bottom: 0,
+  pointerEvents: 'none',
+  forcedColorAdjust: 'none',
+
+  boxShadow: {
+    isFocusVisible: '[inset 2px 0 0 0 var(--spectrum-alias-focus-color), inset -2px 0 0 0 var(--spectrum-alias-focus-color), inset 0 -2px 0 0 var(--spectrum-alias-focus-color), inset 0 2px 0 0 var(--spectrum-alias-focus-color)]',
+    isSelected: {
+      default: '[inset 1px 0 0 0 var(--spectrum-alias-focus-color), inset -1px 0 0 0 var(--spectrum-alias-focus-color), inset 0 -1px 0 0 var(--spectrum-alias-focus-color), inset 0 1px 0 0 var(--spectrum-alias-focus-color)]',
+      isFocusVisible: '[inset 2px 0 0 0 var(--spectrum-alias-focus-color), inset -2px 0 0 0 var(--spectrum-alias-focus-color), inset 0 -2px 0 0 var(--spectrum-alias-focus-color), inset 0 2px 0 0 var(--spectrum-alias-focus-color)]'
+    },
+    forcedColors: {
+      isFocusVisible: '[inset 2px 0 0 0 Highlight, inset -2px 0 0 0 Highlight, inset 0 -2px 0 0 Highlight, inset 0 2px 0 0 Highlight]',
+      isSelected: {
+        default: '[inset 1px 0 0 0 Highlight, inset -1px 0 0 0 Highlight, inset 0 -1px 0 0 Highlight, inset 0 1px 0 0 Highlight]',
+        isFocusVisible: '[inset 2px 0 0 0 Highlight, inset -2px 0 0 0 Highlight, inset 0 -2px 0 0 Highlight, inset 0 2px 0 0 Highlight]'
+      }
+    }
+  }
+});
+
+export const TreeViewItem = (props: SpectrumTreeViewItemProps) => {
+  let {
+    children,
+    childItems,
+    hasChildItems,
+    href
+  } = props;
+
+  let content;
+  let nestedRows;
+  let {renderer} = useTreeRendererContext();
+  // TODO alternative api is that we have a separate prop for the TreeItems contents and expect the child to then be
+  // a nested tree item
```

Comment:
This allows us to reuse the function renderer the user passes to TreeView to render the nested rows